### PR TITLE
Shanice/tfsec to trivy

### PR DESCRIPTION
**Pull request by shanice-skylight**
## Related Issue

Resolves Migrate from tfsec to Trivy #67 

- Per warnings found in the most recent executions of tfsec, the Aqua team has shifted their focus to Trivy. tfsec will continue to be available, but it is unclear just how often it will be kept updated.

## Changes Proposed

- Run Trivy as a check on each PR that gets opened in the repository. This helps identify potential vulnerabilities that might get introduced with each PR.
- Disable the tfsec workflow

## Additional Information

- This update will run Trivy in Repo Mode: This tells Trivy to scan the repository.
- It will scan for Vulnerabilities, Misconfigurations, and Secrets
- This scans the code in the branch in repository mode but if deemed necessary we can have trivy also scan the docker image

## Testing

- Verify trivy scanning results are displayed in the console 
https://github.com/CDCgov/dibbs-cloud/actions/runs/9506240582
 
Screenshot 2024-06-13 at 4 35 05 PM

### CODE REVIEW

**Comment by shanice-skylight** [comment]
The above comment is obsolete, I configured it to show the results in the pipeline console

File: .github/workflows/trivy.yml
Comment (rin-skylight) [inline .github/workflows/trivy.yml:27]: This line seems like it's causing a problem. We have critical vulnerabilities, based on the report, but the check is showing as "passing". Would bumping this to "false" cause it to fail as we would expect?

If not, we need to take a closer look at how this functions, because this doesn't block on critical issues at the moment.

File: .github/workflows/trivy.yml
Comment (rin-skylight) [inline .github/workflows/trivy.yml:25]: We'll need this to scan more than just the `dev` folder. Can we expand the scope to the entirety of `ops`?

**Review by rin-skylight — CHANGES_REQUESTED** [review]

**Comment by rin-skylight** [comment]
As a note: for this ticket to be complete, the `tfsec` workflow will also need to be deleted.

File: .github/workflows/trivy.yml
Comment (shanice-skylight) [reply .github/workflows/trivy.yml]: Done

File: .github/workflows/trivy.yml
Comment (shanice-skylight) [reply .github/workflows/trivy.yml]: Changed ignore-unfixed and added exit-code, now moving forward this check will fail if a critical or high vulnerability is found. https://github.com/CDCgov/dibbs-cloud/actions/runs/9522759633/attempts/1

**Comment by shanice-skylight** [comment]
The `tfsec` workflow is now deleted. I created a [ticket](https://app.zenhub.com/workspaces/dibbs-cloud-enablement-659dde695f06770a91327bda/issues/gh/cdcgov/dibbs-cloud/91) to fix the Critical and High vulnerabilities found in the terraform code

**Review by rin-skylight — APPROVED** [review]
YES! This is wonderful, thank you for all the changes! I've removed `tfsec` as a required check, since that flow is now obsolete. Once this PR is merged, I'll add the `trivy` workflow in its place.

Thank you, @shanice-skylight !

**Comment by rin-skylight** [comment]
You have my approval to merge, even with the failing `trivy` check.

**Pull request merged**
